File: client/src/main/java/commands/pcommands/Info.java
package commands.pcommands;

import collection.CollectionFilter;
import commands.AbstractCommand;
import commands.CommandInfo;
import commands.ExecutionPayload;
import commands.ExecutionResult;

import static commands.ExecutionMode.CLIENT;

public final class Info extends AbstractCommand {
  private final CollectionFilter collectionFilter;

  public Info(CollectionFilter collectionManager) {
    super(CommandInfo.of("info", "Displays general information about collection", true, 0, false, CLIENT));
    this.collectionFilter = collectionManager;
  }

  @Override
  public ExecutionResult execute(ExecutionPayload payload) {
    String collectionInformation = String.format("Collection of Flats: %nInit time: %s%nVersion: %s%nNumber of elements: %s%n", collectionFilter.getCreationDateTime(), collectionFilter.getCollectionVersion(), collectionFilter.getSize());
    return ExecutionResult.valueOf(true, collectionInformation);
  }
}

File: client-ui/domain/src/commonMain/kotlin/ru.moleus.kollector/domain/commands/pcommands/Info.kt
package ru.moleus.kollector.domain.commands.pcommands

import commands.*
import ru.moleus.kollector.domain.collection.CollectionFilter

class Info(private val collectionFilter: CollectionFilter) : AbstractCommand(
    CommandInfo.of(
        "info",
        "Displays general information about collection",
        true,
        0,
        CommandType.OTHER,
        ExecutionMode.CLIENT
    )
) {
    override fun execute(payload: ExecutionPayload): ExecutionResult {
        val collectionInformation = String.format(
            "Collection of Flats: %nInit time: %s%nVersion: %s%nNumber of elements: %s%n",
            collectionFilter.creationDateTime,
            collectionFilter.getCollectionVersion(),
            collectionFilter.size
        )
        return ExecutionResult.valueOf(true, collectionInformation)
    }
}